hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java

```diff
@@ -108,10 +118,23 @@ public FSTableDescriptors(final FileSystem fs, final Path rootdir) {
 
   public FSTableDescriptors(final FileSystem fs, final Path rootdir, final boolean fsreadonly,
     final boolean usecache) {
+    this(fs, rootdir, fsreadonly, usecache, 0);
+  }
+
+  public FSTableDescriptors(final FileSystem fs, final Path rootdir, final boolean fsreadonly,
+    final boolean usecache, final int tableDescriptorParallelLoadThreads) {
     this.fs = fs;
     this.rootdir = rootdir;
     this.fsreadonly = fsreadonly;
     this.usecache = usecache;
+    if (tableDescriptorParallelLoadThreads > 0) {
+      tableDescriptorParallelLoadEnable = true;
+      executor = new ThreadPoolExecutor(tableDescriptorParallelLoadThreads,
```

Comment:
The executor is never shut down after initialization. it's better to shut it down gracefully when the server is stopped. We apply a similar logic for the `consumeExecutor` in `AsyncFSWAL`